api/include/opentelemetry/trace/trace_state.h

```diff
@@ -45,51 +45,154 @@ class TraceState
   class Entry
   {
   public:
-    Entry() noexcept = default;
-
-    // Copy constructor.
-    Entry(const Entry &copy);
-
-    // Creates an Entry for a given key-value pair.
-    Entry(nostd::string_view key, nostd::string_view value) noexcept;
-
-    nostd::string_view GetKey();
-    nostd::string_view GetValue();
+    Entry() : key_(new char[kKeyMaxSize]), value_(new char[kValueMaxSize]){};
+
+    // Copy constructor
+    Entry(Entry &copy)
+    {
+      key_   = StringToPointer(copy.key_.get(), kKeyMaxSize);
+      value_ = StringToPointer(copy.value_.get(), kValueMaxSize);
+    }
+
+    // Assignment operator
+    Entry &operator=(Entry &other)
+    {
+      key_   = StringToPointer(other.key_.get(), kKeyMaxSize);
+      value_ = StringToPointer(other.value_.get(), kValueMaxSize);
+      return *this;
+    }
+
+    // Create an Entry for a given key-value pair.
+    Entry(nostd::string_view key, nostd::string_view value) noexcept
+    {
+      key_   = StringToPointer(key.data(), kKeyMaxSize);
+      value_ = StringToPointer(value.data(), kValueMaxSize);
+    }
+
+    // Get the key associated with this entry.
+    nostd::string_view GetKey() { return nostd::string_view(key_.get()); }
+
+    // Get the value associated with this entry.
+    nostd::string_view GetValue() { return nostd::string_view(value_.get()); }
+
+    // Set the value for this entry. This overrides the previous value.
+    void SetValue(nostd::string_view value)
+    {
+      value_ = StringToPointer(value.data(), kValueMaxSize);
+    }
 
   private:
     // Store key and value as raw char pointers to avoid using std::string.
-    nostd::unique_ptr<const char[]> key_;
-    nostd::unique_ptr<const char[]> value_;
+    std::unique_ptr<const char[]> key_;
+    std::unique_ptr<const char[]> value_;
+
+    // Copy string into a buffer and return a unique_ptr to the buffer.
+    // This is a workaround for the fact that strcpy doesn't accept a const char* destination.
+    nostd::unique_ptr<const char[]> StringToPointer(const char *str, const int kMaxSize)
+    {
+      nostd::unique_ptr<char[]> temp(new char[kMaxSize]);
+      strcpy(temp.get(), str);
+      return nostd::unique_ptr<const char[]>(temp.release());
+    }
   };
 
   // An empty TraceState.
-  TraceState() noexcept : num_entries_(0) {}
+  TraceState() noexcept : entries_(new Entry[kMaxKeyValuePairs]), num_entries_(0) {}
 
   // Returns false if no such key, otherwise returns true and populates value.
-  bool Get(nostd::string_view key, nostd::string_view value) const noexcept { return false; }
+  bool Get(nostd::string_view key, nostd::string_view value) noexcept
```

Comment:
What's the use case for this function? Shouldn't it rather be called `Replace`?

Reply:
I'm not sure, it was defined by @g-easy in https://github.com/open-telemetry/opentelemetry-cpp/pull/42. At the time, there was no `Set()` method defined, and I think the intention was to combine `Set(key, value)` and `Contains(key)` in one function. Now that we have a `Set()` method, I was thinking it would be better to change this to `Get(key)`, which returns the value associated with the key (null if the key does not exist). What are your thoughts?

Reply:
After speaking with the Context team, I realized that I originally misunderstood the purpose of this function. The intention was for the second parameter to be `string_view* value`, and to return `value` directly as an output parameter.

I've updated this function to take in a `string_view& value` and populate that parameter, and I also updated the tests accordingly.

Reply:
@pyohannes Please let me know your thoughts on the new version of this function!